lib/wor/authentication/exceptions.rb

```diff
@@ -1,15 +1,87 @@
 module Wor
   module Authentication
     module Exceptions
-      class InvalidExpirationDaysError < StandardError; end
-      class InvalidMaximumUsefulDaysError < StandardError; end
-      class SubclassMustImplementError < StandardError; end
-      class NoKeyProvidedError < StandardError; end
-      class ExpiredTokenError < StandardError; end
-      class NotRenewableTokenError < StandardError; end
-      class EntityCustomValidationError < StandardError; end
-      class MissingAuthorizationHeader < StandardError; end
-      class InvalidAuthorizationToken < StandardError; end
+      class BaseError < StandardError
+        def message
+          exception_name = self.class.to_s.split('::').last
+          camel_case_splitted = exception_name.split(/(?=[A-Z])/)
+          camel_case_splitted.join(' ')
+        end
+
+        def status_code
```

Comment:
why does the Base error status code is unauthorized ?

Reply:
It's the most common for our scenario. If you fail some validation you should return that so the user knows that she has to log in again